src/plugins/intel_gpu/src/graph/loop.cpp

```diff
@@ -1019,6 +1019,34 @@ void loop_inst::set_memory_in_body_network(cldnn::network::ptr body_network,
     }
 }
 
+void loop_inst::set_memory_in_body_network_from_output(cldnn::network::ptr body_network,
+                const std::shared_ptr<cldnn::primitive_inst>& inst, memory::ptr mem) {
+    if (inst->is_input()) {
+        // in case where memory buffer has been over-allocated by shape predictor, memory layout might be unexpected shape.
+        // so memory layout needs to be re-interprete according to original layout.
+
+            memory::ptr updated_mem = mem;
+            layout impl_layout = inst->get_impl_params()->get_output_layout();
+            OPENVINO_ASSERT(impl_layout.bytes_count() <= updated_mem->get_layout().bytes_count(),
+                            "impl_params layout size(", impl_layout.to_short_string(),
+                            ") should not exceed memory size(", updated_mem->get_layout().to_short_string(), ")");
+            // if (impl_layout.bytes_count() < updated_mem->get_layout().bytes_count()) {
+            //     updated_mem = body_network->get_engine().reinterpret_buffer(*updated_mem, impl_layout);
+            // }
+            inst->set_output_layout(updated_mem->get_layout(), 0);
+            auto& inst_engine = body_network->get_engine();
+            auto new_mem = inst_engine.allocate_memory(updated_mem->get_layout(), updated_mem->get_allocation_type(), true);
+            inst->set_output_memory(new_mem, false, 0);
```

Comment:
you add new memory to the input of body network and then set input data using updated_mem for same input primitive_inst. Why do you allocate new mem for input primitive_inst?

Reply:
Thanks for your advice, code is changed in the new commit.